src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,375 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.Stream.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP
+namespace Microsoft.OData.Json
+{
+    using System;
+    using System.Buffers;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal sealed partial class ODataUtf8JsonWriter
+    {
+        /// <summary>
+        /// Starts a scope for writing a stream value.
+        /// </summary>
+        /// <returns>A stream for writing the stream value.</returns>
+        public Stream StartStreamValueScope()
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            this.WriteItemWithSeparatorIfNeeded();
+
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            this.Flush();
+
+            this.binaryValueStream =  new ODataUtf8JsonWriteStream(this);
+
+            return this.binaryValueStream;
+        }
+
+        /// <summary>
+        /// Ends a scope for writing a stream value.
+        /// </summary>
+        public void EndStreamValueScope()
+        {
+            this.binaryValueStream.Flush();
+            this.binaryValueStream.Dispose();
+            this.binaryValueStream = null;
+            this.Flush();
+
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+        }
+
+        /// <summary>
+        /// Asynchronously starts a scope for writing a stream value.
+        /// </summary>
+        /// <returns>A task representing the asynchronous operation. The task result contains a stream for writing the stream value.</returns>
+        public async Task<Stream> StartStreamValueScopeAsync()
+        {
+            this.WriteSeparatorIfNecessary();
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+
+            this.binaryValueStream = new ODataUtf8JsonWriteStream(this);
+
+            return this.binaryValueStream;
+        }
+
+        /// <summary>
+        /// Asynchronously ends a scope for writing a stream value.
+        /// </summary>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public async Task EndStreamValueScopeAsync()
+        {
+            await this.binaryValueStream.FlushAsync();

```

Comment:
If we flush when we dispose, would it be sufficient to just call Dispose() instead of calling flush then dispose here?